### PR TITLE
feat: mask connection values in CreateConnection and AttachTable.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13714)

### CODE REVIEW

File: src/query/ast/src/ast/statements/statement.rs [hunk 1]
```diff
@@ -283,6 +284,20 @@ impl Statement {
...
             }
+            Statement::AttachTable(attach) => {
+                let mut attach_clone = attach.clone();
+                attach_clone.uri_location.connection = attach_clone.uri_location.connection.mask();
+                format!("{}", Statement::AttachTable(attach_clone))
+            }
+            Statement::CreateConnection(stmt) => {
+                let mut stmt_clone = stmt.clone();
+                stmt_clone.storage_params = stmt_clone
+                    .storage_params
```
Comment (BohuTANG) [inline src/query/ast/src/ast/statements/statement.rs:295]: Impl a debug with mask for storage_params?

Comment (youngsofun) [reply src/query/ast/src/ast/statements/statement.rs]: done, impl in Display, leave normal Debug to developer for debugging

File: src/query/ast/src/ast/statements/connection.rs [hunk 2]
```diff
@@ -48,8 +48,8 @@ impl Display for CreateConnectionStmt {
         }
         write!(f, "{} ", self.name)?;
         write!(f, "STORAGE_TYPE = {} ", self.storage_type)?;
-        for (k, v) in &self.storage_params {
-            write!(f, "{} = {} ", k, v)?;
+        for k in &self.storage_params.keys() {
+            write!(f, "{} = ******** ", k)?;
```
Comment (BohuTANG) [inline src/query/ast/src/ast/statements/connection.rs:52]: Prefre to use https://github.com/datafuselabs/databend/blob/main/src/common/base/src/base/string.rs#L101
`mask_string(v, 3)`

Comment (youngsofun) [reply src/query/ast/src/ast/statements/connection.rs]: done

**Review by BohuTANG — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
```
02_0005_attach_table_read_only:                                         [ FAIL ] - result differs with:
--- /runner/_work/databend/databend/tests/suites/1_stateful/02_query/02_0005_attach_table_read_only.result	2023-11-15 01:47:18.243513031 +0000
+++ /runner/_work/databend/databend/tests/suites/1_stateful/02_query/02_0005_attach_table_read_only.stdout	2023-11-15 01:49:38.959409670 +0000
@@ -41,4 +41,4 @@
 undrop table should work
 0
 show create attach table
-attach_read_only	ATTACH TABLE `default`.`attach_read_only` 'sPLACE_HOLDER://testbucket/admin/PLACE_HOLDER/PLACE_HOLDER/' CONNECTION = ( access_key_id = '********', endpoint_url = '********', secret_access_key = '********' ) READ_ONLY
+attach_read_only	ATTACH TABLE `default`.`attach_read_only` 'sPLACE_HOLDER://testbucket/admin/PLACE_HOLDER/PLACE_HOLDER/' CONNECTION = ( access_key_id = '******min', endpoint_url = '******PLACE_HOLDER', secret_access_key = '******min' ) READ_ONLY
```

**Comment by BohuTANG** [comment]
Order issue:
```
--- /runner/_work/databend/databend/tests/suites/1_stateful/02_query/02_0005_attach_table_read_only.result	2023-11-15 02:24:21.316689990 +0000
+++ /runner/_work/databend/databend/tests/suites/1_stateful/02_query/02_0005_attach_table_read_only.stdout	2023-11-15 02:25:47.008644475 +0000
@@ -41,4 +41,4 @@
 undrop table should work
 0
 show create attach table
-attach_read_only	ATTACH TABLE `default`.`attach_read_only` 's3://testbucket/admin/1/11756/' CONNECTION = ( access_key_id = '******min', endpoint_url = '******900', secret_access_key = '******min' ) READ_ONLY
+attach_read_only	ATTACH TABLE `default`.`attach_read_only` 'sPLACE_HOLDER://testbucket/admin/PLACE_HOLDER/PLACE_HOLDER/' CONNECTION = ( access_key_id = '******min', endpoint_url = '******PLACE_HOLDER', secret_access_key = '******min' ) READ_ONLY
```

**Comment by youngsofun** [comment]
> ATTACH TABLE `default`.`attach_read_only` 's3://testbucket/admin/1/11756/' CONNECTION = ( access_key_id = '******min', endpoint_url = '******900', secret_access_key = '******min' ) READ_ONLY

```
echo "show create table attach_read_only" | $BENDSQL_CLIENT_CONNECT | sed -e 's/[0-9]\+/PLACE_HOLDER/g'
```

 sed on mac is different with linux

**Pull request merged**